src/branch.ts

```diff
@@ -0,0 +1,83 @@
+import * as core from '@actions/core';
+import * as github from '@actions/github';
+
+export interface BranchMatchConfig {
+  headBranch?: string[];
+  baseBranch?: string[];
+}
+
+type BranchBase = 'base' | 'head';
+
+export function toBranchMatchConfig(config: any): BranchMatchConfig {
+  if (!config['head-branch'] && !config['base-branch']) {
+    return {};
+  }
+
+  const branchConfig = {
+    headBranch: config['head-branch'],
+    baseBranch: config['base-branch']
+  };
+
+  if (branchConfig.headBranch) {
+    const patterns = branchConfig.headBranch;
+    if (typeof patterns === 'string') {
+      branchConfig.headBranch = [patterns];
+    }
+  }
+
+  if (branchConfig.baseBranch) {
+    const patterns = branchConfig.baseBranch;
+    if (typeof patterns === 'string') {
+      branchConfig.baseBranch = [patterns];
+    }
+  }
+
+  return branchConfig;
+}
+
+export function getBranchName(branchBase?: BranchBase): string | undefined {
```

Comment:
Could you please clarify why the parameter is optional?

Reply:
I guess that was my own personal bias for the use case that I originally created this PR for and the original implementation - but yeah it doesn't really need to be optional, I'll remove that and just make it always accept an argument

Reply:
I guess that was my own personal bias for the use case that I originally created this PR for and the original implementation - but yeah it doesn't really need to be optional, I'll remove that and just make it always accept an argument